### PR TITLE
Migration from Leap 42.3 to SLE

### PR DESCRIPTION
**Pull request by lnussel**
Automated migration test from Leap 42.3 to SLE

Needs a separate run to generate a text mode image including all updates for Leap. Sample job templates are in the commit.
Only two needles are needed AFAICT, one for the 42.3 bootloader and another one for zypper done due to the different font.

This can serve as template for the mandatory migration feature of 15. The 12 specific parts here can be guarded for that.

- Verification run: https://tortuga.suse.de/tests/311

### CODE REVIEW

File: tests/installation/zdup_prepare_repos.pm [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# SUSE's openQA tests
```
Comment (okurz) [inline tests/installation/zdup_prepare_repos.pm]: I think the splitout of this test module could be a first PR which we can merge more easily.

**Comment by lnussel** [comment]
based on #5881 now

**Comment by lnussel** [comment]
Just to be clear, the migration 15 -> 15 is a mandatory and supported feature. i'm surprised that no openQA test was produced for GA already. So I'm posting this code here with the intention to make it easy for QA to get started with the 15 migration. Should be easy to add the 15 specific bits where I put the 12 specific ones.

**Comment by okurz** [comment]
Are you really surprised? You have followed https://progress.opensuse.org/issues/29098 as well which explains it :)

**Comment by Sunny-sun** [comment]
> Just to be clear, the migration 15 -> 15 is a mandatory and supported feature. i'm surprised that no openQA test was produced for GA already. So I'm posting this code here with the intention to make it easy for QA to get started with the 15 migration. Should be easy to add the 15 specific bits where I put the 12 specific ones.

Migration from leap to SLE is a new feature for SLE15 GA, which had finished implementation in RC2, so QA didn't create automation test case in openQA for SLE15 GA

**Comment by Sunny-sun** [comment]
> Are you really surprised? You have followed https://progress.opensuse.org/issues/29098 as well which explains it :)

Yes, we target to automate migration testing from leap to SLE for SLE15 SP1.

**Comment by rwx788** [comment]
There is #7410 which aims the scenarios with zdup. Not sure if this PR was used as a base.

**Comment by okurz** [comment]
@rwx788 it looks like it :)

**Comment by wjn740** [comment]
Travis report 

"test-merge" failed. There is a little problem in checking code. it will be fixed in #7790.

**Comment by lnussel** [comment]
superseded by #7410, closing

**Pull request closed without merging**